### ISSUE TITLE
Preference Change Listener

### ISSUE THREAD

**Issue by sugarmanz** [MarcinMoskala/PreferenceHolder#3]
I love this implementation of a Preference wrapper, but I find myself using Preference change listeners sometimes and would like to know if this is something you've considered adding functionality for.

**Comment by MarcinMoskala** [comment]
This code looks good, but I am not sure why placing it inside `PreferenceHolder`? It could be placed in any other class as long as the same `SheredPreferences` are used. Maybe instead I should just add an option to register concrete `SheredPreferences` instead of passing Context? Then you could initialize both in Application:
```
val preferences = PreferenceManager.getDefaultSharedPreferences(context)
PreferenceHolder.setPreferences(preferences)
PreferenceListeners.setPreferences(preferences)
```
This solution would be more general.

**Comment by sugarmanz** [comment]
Yeah. That makes a bit more sense. The only reason I would consider including it in this library is it would remove boilerplate code for each app I want to use preference listeners in conjunction with preferenceholder.
